### PR TITLE
Correct TF formatting to exclude LayerNorms from weight decay

### LINKED ISSUE #4360: LayerNorm not excluded from weight decay in TF
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...):
bert-base-cased

Language I am using the model on (English, Chinese ...):
English

The problem arises when using:
* [X] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [X] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1. Add a print statement to `_do_use_weight_decay` in [AdamWeightDecay](https://github.com/huggingface/transformers/blob/master/src/transformers/optimization_tf.py) to see which parameters are actually excluded:
```python
    def _do_use_weight_decay(self, param_name):
        """Whether to use L2 weight decay for `param_name`."""
        if self.weight_decay_rate == 0:
            return False

        if self._include_in_weight_decay:
            for r in self._include_in_weight_decay:
                if re.search(r, param_name) is not None:
                    return True

        if self._exclude_from_weight_decay:
            for r in self._exclude_from_weight_decay:
                if re.search(r, param_name) is not None:
                    print(f"Found: {param_name}")
                    return False
        return True
```
2. run `python examples/text-classification/run_tf_glue.py --model_name_or_path bert-base-cased --task_name mrpc --output_dir temp --logging_dir temp --do_train --overwrite_output_dir --optimizer_name adamw`. 
3. Observe that no weights related to layer norms are printed. 

## Expected behavior

The weights of the layer norms (and the biases) should be printed. 

See for example: https://github.com/google-research/bert/blob/f39e881b169b9d53bea03d2d341b31707a6c052b/optimization.py. 

Based on the fact that no layer norm weights are printed with "layer_norm" simply switching "layer_norm" to "LayerNorm" seems like the easiest change. 

## Environment info

     
- `transformers` version: 2.9.0
- Platform: Darwin-19.4.0-x86_64-i386-64bit
- Python version: 3.7.7
- PyTorch version (GPU?): 1.5.0 (False)
- Tensorflow version (GPU?): 2.2.0 (False)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### PR DESCRIPTION
**Pull request by oliverastrand**
Fixes #4360 

Layer Norm is formated in the wrong way for TensorFlow. This causes it not to be excluded from weight decay in the [run_tf_glue.py](https://github.com/huggingface/transformers/blob/master/examples/text-classification/run_tf_glue.py) script. 

This pr simply formats the string to fit the TensorFlow naming. 

Not sure if you want a test for this? One option would be to check that all elements in `_exclude_from_weight_decay` trigger a regexp match. But seems a bit overkill.

### CODE REVIEW

File: src/transformers/optimization_tf.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def create_optimizer(init_lr, num_train_steps, num_warmup_steps, end_lr=0.0, opt
         beta_1=0.9,
         beta_2=0.999,
         epsilon=1e-6,
-        exclude_from_weight_decay=["layer_norm", "bias"],
+        exclude_from_weight_decay=["LayerNorm", "bias"],
```
Comment (jplu) [inline src/transformers/optimization_tf.py:78]: Good catch!

I suggest instead of replacing the value to add it in case some layers are called `layer_norm` in a model. At the end to have:

`exclude_from_weight_decay=["LayerNorm", "layer_norm", "bias"]

**Comment by jplu** [comment]
Hello!

Thanks for the fix! Just one suggestion above :)

**Comment by oliverastrand** [comment]
Thanks for the feedback! Sounds like a good idea, added that. Are the failing ci tests an issue? Everything passes on my machine.

**Comment by jplu** [comment]
Awesome! LGTM :)

/cc @julien-c and @LysandreJik

**Review by LysandreJik — APPROVED** [review]
